architecture.md

```diff
@@ -1,205 +1,42 @@
 
-- Component source code is fetched from on chain and transpiled from the current JSX to JavaScript (Preact)
-- A new iframe is created to encapsulate the translated JavaScript in a closure used to render the Component inside the outer window application
-- The Component props and DOM tree are serialized and posted to the outer window application to be rendered
-- A parent Component rendering Component children renders a placeholder DOM node (a leaf on the parent Component’s DOM tree) and the outer window application renders a Component iframe container (with props) for each child
-- Component iframe containers may expose methods to be invoked from the outer window application via message posting, either as event handlers or indirect callbacks originating from other Components proxied by the outer window application
-- Components are re-rendered in response to callbacks in the outer window, and re-render their children in turn
+### BOS Component Loading
+BOS Components are loaded in BOS Web Engine by way of a client-side pipeline, responsible for BOS Component source
+fetching, JSX transpilation, and Component composition. A web worker services requests from the outer application to
+load new Components, caching the Component source and transpiled wrapper function. The worker responds with the wrapped
+source code, which is used to initialize containers with iframes.
 
-## Module Loading
+![source-compile-container](./assets/source-compile-container.png)
+_High-level overview of the flow from BOS Component source to Component container._
 
-At a high level, module loading is as follows:
+### Component Containers
+Containers are abstractions around sandboxed iframes, responsible for managing the lifecycle of a single root Component.
+The outer application renders a container for each root Component, rendering the container's hidden iframe and executing
+the container code in the iframe's `srcDoc` field. Once initialized, the container will make a render request to the outer
+application with a message that includes the serialized Component DOM. 
 
-1. Fetch Component’s JSX source
-2. Transpile into Preact
-3. Render Component Preact component in iframe
-   1. Serialize DOM subtree & props
-   2. Build set of child Components to fetch, creating a unique identifier for each
-   3. Post message to outer window
-      1. Render serialized DOM subtree (including DOM placeholders for child Components) in outer window
-      2. Initialize iframes for child Components (each one forking new workflow starting from #1)
+A root Component is only re-rendered in the outer application when the container requests it explicitly. Render requests
+are made when:
+- the container has finished initializing
+- state has changed within the Component (e.g. from a `useState` setter)
+- the root Component's `props` have changed, i.e. its parent Component re-rendered
```

Comment:
can this be simplified to say that it's based on when the Preact engine inside the iframe runs the render function again and detects dom changes, then give these two points as examples of when that frequently happens?